compiler/sem/sem.nim

```diff
@@ -474,52 +478,67 @@ proc tryConstExpr(c: PContext, n: PNode): PNode =
   c.config.errorMax = high(int) # `setErrorMaxHighMaybe` not appropriate here
 
   result = evalConstExpr(c.module, c.idgen, c.graph, e)
-  if result == nil or result.kind in {nkEmpty, nkError}:
+  case result.kind
+  of nkEmpty, nkError:
     result = nil
   else:
-    result = fixupTypeAfterEval(c, result, e)
+    discard
 
   c.config.errorCounter = oldErrorCount
   c.config.errorMax = oldErrorMax
   c.config.m.errorOutputs = oldErrorOutputs
 
-proc semConstExpr(c: PContext, n: PNode): PNode =
-  addInNimDebugUtils(c.config, "semConstExpr", n, result)
+proc evalConstExpr(c: PContext, n: PNode): PNode =
+  ## Tries to turn the expression `n` into AST that represents a concrete
+  ## value. If this fails, an `nkError` node is returned
+  addInNimDebugUtils(c.config, "evalConstExpr", n, result)
+  assert not n.isError
+
+  # this happens when the overloadableEnums is enabled. We short-circuit
+  # evaluation in this case, as neither ``vmgen`` nor ``semfold`` know what to
+  # do with the sym-choice
+  # XXX: this doesn't seem like a good idea. Maybe report an error instead?
```

Comment:
This silly bit is in const, let, var, initialization expression analysis, as well. I can't help but feel that should have been handled earlier. I just can't picture the solution quite yet. Something like symbolic type based evaluation should have been done which either resolves the choice or all this is in a higher level context (eg: generic), where ambiguity isn't an issue.

The levels thing is taking a bunch of my brain space and my mind is only trying to see a solution through that lens at the moment. 😬